src/server.py

```diff
@@ -1,8 +1,301 @@
+    return datetime.now().year
+
+def current_sem() -> str:
+    """Gets the current semester, to be used as a simple default for /courses/  """
+    return "Semester 1" if datetime.now().month <= 6 else "Semester 2" # Not the most bulletproof logic lol
+
+def meeting_date_convert(raw_date: str) -> dict[str]:
+    """Converts the date format given in the meetings to "MM-DD"
+
+    Args:
+        raw_date (str): The given meeting date in the format of "DD {3-char weekday} - DD {3-char weekday}"
+
+    Returns:
+        formatted_date (dict[str]): The formatted meeting date in the format of "MM-DD" in a dict containing 
+        the "start" and "end" keys and their corresponding dates
+    """
+    
+    months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    start, end = raw_date.split(" - ")
+    
+    start_d, start_m = start.split()
+    start_m = str(months.index(start_m) + 1).zfill(2)
+    
+    end_d, end_m = end.split()
+    end_m = str(months.index(end_m) + 1).zfill(2)
+    
+    formatted_date = {
+        "start": f"{start_m}-{start_d.zfill(2)}",
+        "end": f"{end_m}-{end_d.zfill(2)}"
+    }
+    return formatted_date
+
+def meeting_time_convert(raw_time: str) -> str:
+    """Converts the time given in meetings to "HH:mm"
+
+    Args:
+        raw_time (str): The given meeting time in the format of "H{am/pm}"
+
+    Returns:
+        formatted_time (str): The formatted meeting time in the format of "HH:mm"
+    """
+    
+    period = raw_time[-2:]
+    hour = int(raw_time.replace(period, ""))
+    
+    if period.lower() == "pm":
+        hour += 12
+    
+    hour = str(hour).zfill(2)
+    formatted_time = f"{hour}:00"
+    return formatted_time
+
+def parse_requisites(raw_requisites: str) -> Union[dict[str], None]:
+    """Takes in a string of -requisites and returns a dict containing the original string, and a list of the parsed-out subjects
+
+    Args:
+        raw_requisites (str): The raw string containing a list of -requisites, usually in the format of 
+        "COMP SCI 1103, COMP SCI 2202, COMP SCI 2202B" as an example
+
+    Returns:
+        parsed_requisites (Union[dict[str], None]): Either a dict with "desc"s value being the original -requisites string and 
+        "subjects" being a list of the parsed -requisites, or None if raw_requisites is None
+    """
+    
+    if not raw_requisites:
+        return None
+    
+    # Note: This regex pattern was generated by Claude 3.5 Sonnet, works fine in my testing though
+    pattern = r'\b([A-Z]+(?:\s+[A-Z]+)*)\s+(\d{4}\w*)\b'
+    matched_subjects = [" ".join(match) for match in re.findall(pattern, raw_requisites)]
+    
+    parsed_requisites = None
+    if len(matched_subjects) > 0:
+        parsed_requisites = {
+            "desc": raw_requisites,
+            "subjects": matched_subjects
+        }
+    
+    return parsed_requisites
+
+def get_term_number(year: int, term: str) -> int:
```

Comment:
If we plan to support more terms (such as trimesters and quadmesters then this should be changed (we'd want to create an issue highlighting this).

@jsun969 thoughts?

Reply:
yea we do, @Sebby37 could you add all terms

Reply:
It already supports all terms listed on [the CoursePlanner API](https://courseplanner-api.adelaide.edu.au/api/course-planner-query/v1/?target=/system/TERMS/queryx&virtual=Y&year_from=0&year_to=9999), but I added sem1 and sem2 as aliases to Semester 1 and Semester 2 (although I probably wasn't very clear on that). Did we want to keep those aliases or strictly enforce the syntax used in the API?

Reply:
I think @jsun969 wanted to have the aliases for all the terms.

Reply:
Just added a function to convert all term aliases to their API names. I've included aliases for all the different kinds of terms past and present I could find to achieve maximum compatibility. Here they are below, let me know if any of them need changing:
"sem": "Semester",
"elc": "ELC Term",
"tri": "Trimester",
"term": "Term",
"ol": "Online Teaching Period",
"melb": "Melb Teaching Period",
"fast": "Fast Track",
"pce": "PCE Term",
"summer": "Summer School",
"winter": "Winter School"

Reply:
nice this looks good